client/js/components/user/CustomerSettings/CustomerSettingsBranding.vue

```diff
@@ -90,6 +97,59 @@ export default {
     branding: {
       required: true,
       type: Object
+    },
+    brandingId: {
+      required: true,
+      type: String
+    }
+  },
+
+  data () {
+    return {
+      uploadedFileId: '',
+      logoDelete: false
+    }
+  },
+
+  computed: {
+    ...mapState('branding', {
+      brandingList: 'items'
+    })
+  },
+
+  methods: {
+    ...mapActions('branding', {
+      fetchBranding: 'list',
+      saveBranding: 'save'
+    }),
+
+    ...mapMutations('branding', {
+      updateBranding: 'setItem'
+    }),
+
+    getFileId (file) {
+      this.uploadedFileId = file.fileId
+    },
+
+    saveBrandingSettings () {
+      const payload = {
+        id: this.brandingId,
+        type: 'Branding',
+        attributes: {
+          ...this.brandingList[this.brandingId].attributes
+        },
+        relationships: {
+          logo: {
+            data: this.logoDelete ? null : { id: this.uploadedFileId, type: 'file' }
+          }
+        }
+      }
```

Comment:
a general question: Do we want to handle the save actions in the subcomponents like here?or in the parent component like there: client/js/components/user/CustomerSettings/CustomerSettings.vue:503 ?

Reply:
Generelly I would prefer to have save actions in the parent component and only have the emits for the save action in the subcomponents